### PR TITLE
Python ticketing example

### PR DESCRIPTION
**Pull request by gvdongen**
Implemented the ticket reservation example in Python. This needs to be moved to the Tour as soon as we have it online.

### CODE REVIEW

File: patterns-use-cases/ticket-reservation/ticket-reservation-python/example/checkout_service.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    total_price = len(order['tickets']) * 40
+
+    idempotency_key = await ctx.run("idempotency_key", lambda: str(uuid.uuid4()))
+
+    async def pay():
+        return await payment_client.call(idempotency_key, total_price)
+    success = await ctx.run("payment", pay)
+
+    if success:
+        await ctx.run("send_success_email", lambda: email_client.notify_user_of_payment_success(order['user_id']))
```
Comment (slinkydeveloper) [inline patterns-use-cases/ticket-reservation/ticket-reservation-python/example/checkout_service.py:32]: mh wondering if the run name should be optional...

**Review by slinkydeveloper — COMMENTED** [review]
LGTM, perhaps shall I merge my PR first so we add yours to CI? https://github.com/restatedev/examples/pull/173/files#diff-fa14c08b9b407a46139ea5bac66f1a8cf75fdd27a29a126b9bfef7fa6d3ec7ea

File: patterns-use-cases/ticket-reservation/ticket-reservation-python/example/checkout_service.py (hunk 1 above)
Comment (gvdongen) [reply patterns-use-cases/ticket-reservation/ticket-reservation-python/example/checkout_service.py:32]: I would like that

File: .github/workflows/pre-release.yml [hunk 2]
```diff
@@ -67,6 +71,20 @@ jobs:
...
+      # Bump Python SDK
+      - uses: actions/checkout@v3
+        if: github.event.inputs.sdkPythonVersion != ''
+      - uses: actions/setup-python@v5
+        if: github.event.inputs.sdkPythonVersion != ''
+        with:
+          python-version: "3.12"
+      - name: Bump Python SDK
+        if: github.event.inputs.sdkPythonVersion != ''
+        run: ./.tools/update_python_examples.sh ${{ inputs.sdkPythonVersion }}
```
Comment (slinkydeveloper) [inline .github/workflows/pre-release.yml:83]: you need to add at the bottom of this file the `requirements.txt` to `add-paths`

**Review by slinkydeveloper — CHANGES_REQUESTED** [review]

**Review by slinkydeveloper — COMMENTED** [review]
lgtm just fix those two issues

**Comment by slinkydeveloper** [comment]
Can you also add python to the main README? sorry i forgot about this in the previous PR

File: patterns-use-cases/ticket-reservation/ticket-reservation-python/README.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+## Running the example
+
+To set up the example, use the following sequence of commands.
+
+Setup the virtual env:
+
+```shell
+python3 -m venv .venv
+source venv/bin/activate
```
Comment (slinkydeveloper) [inline patterns-use-cases/ticket-reservation/ticket-reservation-python/README.md:17]:
```suggestion
source .venv/bin/activate
```

File: patterns-use-cases/ticket-reservation/ticket-reservation-python/example/cart_object.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+
+@cart.handler()
+async def checkout(ctx: ObjectContext) -> bool:
+    tickets = await ctx.get("tickets") or []
+
+    if len(tickets) == 0:
+        return False
+
+    success = await ctx.service_call(handle, arg={'user_id': ctx.key(),
```
Comment (slinkydeveloper) [inline patterns-use-cases/ticket-reservation/ticket-reservation-python/example/cart_object.py:33]: To improve readibility, i suggest to do import renaming of other handlers. e.g.

```
from example.checkout_service import handle as checkout_handle
```

And then here:

```suggestion
    success = await ctx.service_call(checkout_handle, arg={'user_id': ctx.key(),
```

File: .github/workflows/pre-release.yml (hunk 2 above)
Comment (Deeepak127) [reply .github/workflows/pre-release.yml:83]: Hi

**Pull request merged**